include/seqan3/core/configuration/configuration.hpp

```diff
@@ -196,6 +162,58 @@ class configuration : public std::tuple<configs_t...>
     /*!\name Modifiers
      * \{
      */
+
+    /*!\brief Returns a new configuration by appending the given configuration to the current one.
+     *
+     * \tparam other_configuration_t Another configuration type or configuration element type; each configuration
+     *                               element must model seqan3::detail::config_element_pipeable_with each of the
+     *                               configurations elements of the current configuration.
+     *
+     * \param[in] other_config The other configuration to append to the current one.
+     *
+     * \returns A new configuration containing the appended configuration elements.
+     *
+     * \details
+     *
+     * This function generates a new configuration object containing the appended configuration elements. The current
+     * configuration will not be modified.
+     */
+    template <typename other_configuration_t>
+    //!\cond
+        requires (is_config_element_combineable_v<configs_t, std::remove_cvref_t<other_configuration_t>> && ...)
+    //!\endcond
+    constexpr auto append(other_configuration_t && other_config) const
+    {
+        if constexpr (detail::config_element<std::remove_cvref_t<other_configuration_t>>)
+        {
+            return configuration<configs_t..., std::remove_cvref_t<other_configuration_t>>
+            {
+                std::tuple_cat(static_cast<base_type>(*this),
+                               std::tuple{std::forward<other_configuration_t>(other_config)})
+            };
+        }
+        else
+        {
+            using other_base_t = typename std::remove_cvref_t<other_configuration_t>::base_type;
+            using other_base_maybe_const_t =
+                std::conditional_t<std::is_const_v<std::remove_reference_t<other_configuration_t>>,
+                                   std::add_const_t<other_base_t>,
+                                   other_base_t>;
+            using fwd_other_base_t = std::conditional_t<std::is_lvalue_reference_v<other_configuration_t>,
+                                                        std::add_lvalue_reference_t<other_base_maybe_const_t>,
+                                                        std::add_rvalue_reference_t<other_base_maybe_const_t>>;
```

Comment:
This took me a while.

Maybe add a comment "We transfer the type qualifier, like &, &&, const, and any combination of them, from other_configuration_t to ::base_type."

Reply:
ok, yes. That seems plausible :)